src/FwdState.cc

```diff
@@ -121,8 +121,10 @@ void
 FwdState::closeServerConnection(const char *reason)
 {
     debugs(17, 3, "because " << reason << "; " << serverConn);
-    comm_remove_close_handler(serverConn->fd, closeHandler);
-    closeHandler = NULL;
+    if (closeHandler) {
+        comm_remove_close_handler(serverConn->fd, closeHandler);
+        closeHandler = NULL;
```

Comment:
```suggestion
        closeHandler = nullptr;
```

Reply:
I did not do this fix. The new code does not touch the closeServerConnection method.